instrumentation/camel-2.20/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsConnector.java

```diff
@@ -0,0 +1,171 @@
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.BucketNotificationConfiguration;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.QueueConfiguration;
+import com.amazonaws.services.s3.model.S3Event;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.SetBucketNotificationConfigurationRequest;
+import com.amazonaws.services.sns.AmazonSNSAsyncClient;
+import com.amazonaws.services.sns.model.CreateTopicResult;
+import com.amazonaws.services.sqs.AmazonSQSAsyncClient;
+import com.amazonaws.services.sqs.model.GetQueueAttributesRequest;
+import com.amazonaws.services.sqs.model.PurgeQueueRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import java.util.Collections;
+import java.util.EnumSet;
+import org.elasticmq.rest.sqs.SQSRestServer;
+import org.elasticmq.rest.sqs.SQSRestServerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class AwsConnector {
+  private static final Logger logger = LoggerFactory.getLogger(AwsConnector.class);
+  private AmazonSQSAsyncClient sqsClient;
+  private AmazonS3Client s3Client;
+  private AmazonSNSAsyncClient snsClient;
+  private SQSRestServer sqsRestServer;
+
+  static AwsConnector elasticMq() {
+    AwsConnector awsConnector = new AwsConnector();
+    int sqsPort = PortUtils.findOpenPort();
+    awsConnector.sqsRestServer =
+        SQSRestServerBuilder.withPort(sqsPort).withInterface("localhost").start();
+
+    AWSStaticCredentialsProvider credentials =
+        new AWSStaticCredentialsProvider(new BasicAWSCredentials("x", "x"));
+    AwsClientBuilder.EndpointConfiguration endpointConfiguration =
+        new AwsClientBuilder.EndpointConfiguration("http://localhost:" + sqsPort, "elasticmq");
+    awsConnector.sqsClient =
+        (AmazonSQSAsyncClient)
+            AmazonSQSAsyncClient.asyncBuilder()
+                .withCredentials(credentials)
+                .withEndpointConfiguration(endpointConfiguration)
+                .build();
+
+    return awsConnector;
+  }
+
+  static AwsConnector liveAws() {
+    AwsConnector awsConnector = new AwsConnector();
+
+    awsConnector.sqsClient =
+        (AmazonSQSAsyncClient)
+            AmazonSQSAsyncClient.asyncBuilder().withRegion(Regions.US_EAST_1).build();
+
+    awsConnector.s3Client =
+        (AmazonS3Client) AmazonS3Client.builder().withRegion(Regions.US_EAST_1).build();
+
+    awsConnector.snsClient =
+        (AmazonSNSAsyncClient)
+            AmazonSNSAsyncClient.asyncBuilder().withRegion(Regions.US_EAST_1).build();
+
+    return awsConnector;
+  }
+
+  void createBucket(String bucketName) {
+    logger.info("Create bucket  " + bucketName);
```

Comment:
Nit: you can use slf4j placeholders instead of gluing strings manually
```suggestion
    logger.info("Create bucket  {}", bucketName);
```